evennia/contrib/events/README.md

```diff
@@ -0,0 +1,646 @@
+
+```python
+from evennia.contrib.events.typeclasses import EventCharacter
+
+class Character(EventCharacter):
+
+    # ...
+```
+
+You should do the same thing for your rooms, exits and objects.  Note that the event system works by overriding some hooks.  Some of these features might not be accessible in your game if you don't call the parent methods when overriding hooks.
+
+## Using the `@event` command
+
+The event system relies, to a great extent, on its `@event` command.  Who can execute this command, and who can do what with it, will depend on your set of permissions.
+
+The event system can be used on most Evennia objects, mostly typeclassed objects (excluding players).  The first argument of the `@event` command is the name of the object you want to edit.  It can also be used to know what event types are available for this specific object.
+
+### Examining events and event types
+
+To see the event types connected to an object, use the `@event` command and give the name or ID of the object to examine.  For instance, @event here` to examine the event types on your current location.  Or `@event self` to see the event types on yourself.
+
+This command will display a table, containing:
+
+- The name of each event type (trigger) in the first column.
+- The number of events of this name, and the number of total lines of these events in the second column.
+- A short help to tell you when the event is triggered in the third column.
+
+If you execute `@event #1` for instance, you might see a table like this:
+
+```
++------------------+---------+-----------------------------------------------+
+| Event name       |  Number | Description                                   |
++~~~~~~~~~~~~~~~~~~+~~~~~~~~~+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~+
+| can_delete       |   0 (0) | Can the character be deleted?                 |
+| can_move         |   0 (0) | Can the character move?                       |
+| can_part         |   0 (0) | Can the departing character leave this room?  |
+| delete           |   0 (0) | Before deleting the character.                |
+| greet            |   0 (0) | A new character arrives in the location of    |
+|                  |         | this character.                               |
+| move             |   0 (0) | After the character has moved into its new    |
+|                  |         | room.                                         |
+| puppeted         |   0 (0) | When the character has been puppeted by a     |
+|                  |         | player.                                       |
+| time             |   0 (0) | A repeated event to be called regularly.      |
+| unpuppeted       |   0 (0) | When the character is about to be un-         |
+|                  |         | puppeted.                                     |
++------------------+---------+-----------------------------------------------+
+```
+
+### Creating a new event
```

Comment:
This is, as said, unclear because the "event" is the event sent by the engine. like 'move' or 'puppeted'. What you are describing here is how to attach a callback to the event, for doing something when that event fires.